app/services/govuk_emails/monitor.rb

```diff
@@ -8,33 +13,52 @@ def initialize(scheduled_mail_id)
       @scheduled_mail = ScheduledMailing.find(scheduled_mail_id)
     end
 
+    # IMPORTANT: do not use `response#status` to test whether to send email
+    # to avoid a race-condition resulting in multiple emails being sent
     def call
-      status = GovukEmails::Email.new(@scheduled_mail.govuk_message_id).status
-      @scheduled_mail.update!(status:)
-      send_undeliverable_alerts if status.in?(ScheduledMailing::FAILURE_STATUSES)
+      scheduled_mail.update!(status: response.status)
+
+      send_undeliverable_alerts if scheduled_mail.status.in?(ScheduledMailing::FAILURE_STATUSES)
     end
 
   private
 
+    def response
+      @response ||= GovukEmails::Email.new(scheduled_mail.govuk_message_id)
+    end
+
     def send_undeliverable_alerts
       return unless HostEnv.production?
 
-      AlertManager.capture_message("Unable to deliver mail to #{@scheduled_mail.addressee} - ScheduledMailing record #{@scheduled_mail.id}")
+      AlertManager.capture_message("Unable to deliver mail to #{scheduled_mail.addressee} - ScheduledMailing record #{scheduled_mail.id}")
+
+      send_undeliverable_to_apply_team!
+      send_undeliverable_citizen_start_email_to_provider!
+    end
+
+    def send_undeliverable_to_apply_team!
       ScheduledMailing.send_now!(mailer_klass: UndeliverableEmailAlertMailer,
                                  mailer_method: :notify_apply_team,
                                  legal_aid_application_id: legal_aid_application.id,
                                  addressee: Rails.configuration.x.support_email_address,
-                                 arguments: [@scheduled_mail.id])
-
-      notify_provider_email_failed if @scheduled_mail.mailer_method == "citizen_start_email"
+                                 arguments: [scheduled_mail.id])
     end
 
-    def notify_provider_email_failed
+    def send_undeliverable_citizen_start_email_to_provider!
+      return unless scheduled_mail.mailer_method == "citizen_start_email" && scheduled_mail.status == "permanent-failure"
+
       ScheduledMailing.send_now!(mailer_klass: UndeliverableEmailAlertMailer,
                                  mailer_method: :notify_provider,
                                  legal_aid_application_id: legal_aid_application.id,
                                  addressee: provider_email_or_support,
                                  arguments: mailer_args)
+
+      cancel_citizen_start_email_and_reminders!
+    end
+
+    def cancel_citizen_start_email_and_reminders!
+      scheduled_mail.cancel!
+      legal_aid_application.scheduled_mailings.where(mailer_klass: "SubmitCitizenFinancialReminderMailer").map(&:cancel!)
```

Comment:
Do we need to return the array of outputs from `cancel!`?

If not, should we use `.each(&:cancel!)` instead?